packages/omgidl-serialization/src/MessageReader.ts

```diff
@@ -261,15 +260,7 @@ export class MessageReader<T = unknown> {
           // last arrayLengths length is handled in deserializer. It returns an array
           return readNestedArray(typedArrayDeserializer, arrayLengths.slice(0, -1), 0);
         } else {
-          // Special case can happen where the emHeader is not divisible by the typeLength and there is a remainder of 4 bytes.
-          // 4 bytes of 0's are printed after the header length
-          // This can happen for 8 byte typeLength types.
```

Comment:
was this just wrong? or where is the sequence length getting read now?

Reply:
Yeah this was just wrong because I didn't know about the lengthCode caveat. Previously I was always using emHeader size instead of sequenceLength. This case would've arisen for when the lengthCode was <5 and the object size was less than the size of the type. So there was an emHeader object size of 4 and a sequence length. Now that I'm accounting for length code, I will read the sequenceLength earlier in the array handling because the emHeaderSize isn't being used.